streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -269,10 +269,16 @@ void closeRevokedSuspendedTasks() {
 
         if (exception != null) {
             throw exception;
-        } else if (!(active.isEmpty() && assignedActiveTasks.isEmpty() && changelogReader.isEmpty())) {
+        } else if (!(active.isEmpty())) {
+            log.error("The set of active tasks was non-empty: {}", active);
+            throw new IllegalStateException("TaskManager found leftover active task state after closing all zombies");
```

Comment:
If we throw there, we don't check the other conditions and might miss some logs. Might be better to only log for each condition and throw only after each condition was checked (and if at least one condition hit)?

Reply:
Yeah actually that would be significantly more helpful for debugging, I'll make it happen